topiary/rna/common.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2015. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def infer_separator(filename):
+    """
+    Given a file which contains data separated by one of the following:
+        - commas
+        - tabs
+        - spaces
+    Return the most likely separator by sniffing the first 1000 bytes
+    of the file's contents.
+    """
+    with open(filename, "r") as f:
+        # read first thousand bytes of the file which should contain at
+        # least one instance of the field separator
+        substring = f.read(1000)
+        comma_counts = substring.count(",")
+        tab_counts = substring.count("\t")
+        if comma_counts > tab_counts:
+            return ","
+        elif tab_counts > 0:
```

Comment:
This seems like it could be pretty error-prone, especially when all you need is a single tab to infer tab-delim.

What if, after inferring, you try splitting each of those 1000 lines by that delim and checking that the # of elements per line is the same for all 1000? And if not just error?